trollmoves/tests/test_ssh_server.py

```diff
@@ -175,6 +175,32 @@ def test_scp_open_connection_socket_timeout_exception(self, mock_sshclient_conne
         finally:
             logger.removeHandler(stream_handler)
 
+    @patch('paramiko.SSHClient', autospec=True)
+    def test_scp_open_connection_backup_targets(self, mock_sshclient):
+        """Check scp get_connection using backup targets."""
+        from trollmoves.movers import ScpMover
+        stream_handler = logging.StreamHandler(sys.stdout)
+        logger.addHandler(stream_handler)
+        logger.setLevel(logging.INFO)
+
+        mocked_client = MagicMock(side_effect=socket.timeout)
+        mock_sshclient.return_value.connect.side_effect = mocked_client
+
+        scp_mover = ScpMover(self.origin, self.destination_no_port,
+                             attrs={'ssh_connection_timeout': 1,
+                                    'backup_targets': ['backup_host1',
+                                                       'backup_host2']})
+        try:
+            with self.assertLogs(logger, level=logging.INFO) as lc, self.assertRaises(IOError):
+                scp_mover.open_connection()
+            self.assertIn(("SSH connection timed out:"), lc.output[0])
+            self.assertIn(("Changing destination to backup target: backup_host1"), lc.output[3])
+            self.assertIn(("Changing destination to backup target: backup_host2"), lc.output[7])
```

Comment:
Now that we are using pytest, the preferred way to do this is
`assert "SSH connection timed out:" in lc.output[0]`

Also caplog could be used, but no rush for that.